web-app/src/app/screens/Feeds/index.tsx

```diff
@@ -0,0 +1,265 @@
+  TextField,
+  Typography,
+} from '@mui/material';
+import { Search } from '@mui/icons-material';
+
+import '../../styles/SignUp.css';
+import '../../styles/FAQ.css';
+import { selectUserProfile } from '../../store/profile-selectors';
+import { useAppDispatch } from '../../hooks';
+import { loadingFeeds } from '../../store/feeds-reducer';
+import { selectFeedsData } from '../../store/feeds-selectors';
+import {
+  type GTFSFeedType,
+  type GTFSRTFeedType,
+} from '../../services/feeds/utils';
+import SearchResultItem from './SearchResultItem';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+const SEARCH_LIMIT_INCREASE = 20;
+
+const getDataTypeParamFromSelectedFeedTypes = (
+  selectedFeedTypes: Record<string, boolean>,
+): 'gtfs' | 'gtfs_rt' | undefined => {
+  let dataTypeQueryParam: 'gtfs' | 'gtfs_rt' | undefined;
+  if (selectedFeedTypes.gtfs && !selectedFeedTypes.gtfs_rt) {
+    dataTypeQueryParam = 'gtfs';
+  } else if (!selectedFeedTypes.gtfs && selectedFeedTypes.gtfs_rt) {
+    dataTypeQueryParam = 'gtfs_rt';
+  } else if (!selectedFeedTypes.gtfs && !selectedFeedTypes.gtfs_rt) {
+    dataTypeQueryParam = undefined;
+  }
+  return dataTypeQueryParam;
+};
+
+export default function Feed(): React.ReactElement {
+  const navigate = useNavigate();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchLimit, setSearchLimit] = useState(10);
+  const [selectedFeedTypes, setSelectedFeedTypes] = useState({
+    gtfs: true,
+    gtfs_rt: true,
+  });
+
+  const user = useSelector(selectUserProfile);
+  const dispatch = useAppDispatch();
+  const feedsData = useSelector(selectFeedsData);
+
+  const handleSearch = (): void => {
+    const searchQuery = searchParams.get('q') ?? undefined;
+    if (
+      user?.accessToken !== undefined &&
+      searchQuery !== undefined &&
+      searchQuery.trim() !== ''
+    ) {
+      setSearchLimit(10);
+      dispatch(
+        loadingFeeds({
+          accessToken: user?.accessToken,
+          params: {
+            query: {
+              limit: searchLimit,
+              offset: 0,
+              search_query: searchQuery,
+            },
+          },
+        }),
+      );
+    }
+  };
+
+  useEffect(() => {
+    const searchQuery = searchParams.get('q') ?? undefined;
+    const timer = setTimeout(() => {
+      if (
+        user?.accessToken !== undefined &&
+        searchQuery !== undefined &&
+        searchQuery.trim() !== ''
+      ) {
+        dispatch(
+          loadingFeeds({
+            accessToken: user?.accessToken,
+            params: {
+              query: {
+                limit: searchLimit,
+                search_query: searchQuery,
+                data_type:
+                  getDataTypeParamFromSelectedFeedTypes(selectedFeedTypes),
+              },
+            },
+          }),
+        );
+      }
+    }, 400);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [searchLimit, searchParams, selectedFeedTypes]);
```

Comment:
Let's add here the login event status. This can remove the need of the timer. Example code: https://github.com/MobilityData/mobility-feed-api/blob/8c92065d942993fcdcb5531f8b313fd21da723d3/web-app/src/app/screens/Feed/index.tsx#L73

Reply:
I see you added this in main and I merged it in. The timer is needed to prevent search API calls on each keystroke, but this is probably better done by debouncing which I can add.

Reply:
The _isAuthenticatedOrAnonymous_ is needed to avoid calls to the API that returned 403 as no users/anonymous cannot be in the initial page rendering state. I got the idea of the timer now, thanks. We were thinking of a different user interaction, but the design will be available for review later this week.